✔ Uploaded to EAS --> Unexpected token D in JSON at position 1

### Build/Submit details page URL

https://expo.dev/accounts/kernalan/projects/lifesage

### Summary

Running `eas build --platform ios --profile development` triggers a json parsing error as soon as the archive hits the EAS server: 

### Managed or bare?

Managed

### Environment

```
  expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 11 10.0.22631
    Binaries:
      Node: 18.18.0 - C:\Program Files\nodejs\node.EXE
      npm: 10.8.1 - ~\AppData\Roaming\npm\npm.CMD
    npmPackages:
      expo: ~51.0.31 => 51.0.31
      expo-router: ~3.5.23 => 3.5.23
      react: ^18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.74.5 => 0.74.5
      react-native-web: ~0.19.10 => 0.19.12
    Expo Workflow: managed
```
```
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!
```

### Error output

```
Compressing project files and uploading to EAS Build. Learn more: https://expo.fyi/eas-build-archive
✔ Uploaded to EAS
Unexpected token D in JSON at position 1
    Error: build command failed.
```

### Reproducible demo or steps to reproduce from a blank project

1. Run `eas build --platform ios --profile development`
2. Encounter the error

Here was the problematic `app.config.json`:

```
const path = require("path");
const dotenv = require("dotenv");

module.exports = () => {
  // Determine which .env file to use
  const envPath = process.env.EAS_BUILD
    ? ".env.production"
    : process.env.NODE_ENV === "production"
    ? ".env.production"
    : ".env.development";

  console.log(`[DEBUG] Attempting to load environment file: ${envPath}`);

  const result = dotenv.config({ path: path.resolve(__dirname, envPath) });

  if (result.error) {
    console.error(`[ERROR] Failed to load ${envPath}:`, result.error);
  } else {
    console.log(`[INFO] Successfully loaded ${envPath}`);
  }

  console.log("[DEBUG] Environment variables in app.config.js:", {
    API_BASE_URL: process.env.API_BASE_URL,
    NODE_ENV: process.env.NODE_ENV,
  });

  const requiredVars = [
    "FIREBASE_API_KEY",
    "FIREBASE_AUTH_DOMAIN",
    "FIREBASE_PROJECT_ID",
    "FIREBASE_STORAGE_BUCKET",
    "FIREBASE_MESSAGING_SENDER_ID",
    "FIREBASE_APP_ID",
    "API_BASE_URL",
  ];

  requiredVars.forEach((varName) => {
    if (!process.env[varName]) {
      throw new Error(`Missing required environment variable: ${varName}`);
    }
  });

  return {
    expo: {
      name: "lifesage",
      scheme: "lifesage",
      slug: "lifesage",
      version: "1.0.1",
      orientation: "portrait",
      icon: "./assets/icon.png",
      userInterfaceStyle: "light",
      splash: {
        image: "./assets/splash.png",
        resizeMode: "contain",
        backgroundColor: "#ffffff",
      },
      assetBundlePatterns: ["**/*"],
      ios: {
        supportsTablet: true,
        bundleIdentifier: "com.kernalan.lifesage",
        buildNumber: "1",
        googleServicesFile: "./GoogleService-Info.plist",
        entitlements: {
          "aps-environment": "production",
        },
        podfileProperties: {
          "use_modular_headers!": true,
          "use_frameworks!": "static",
        },
        usesAppleSignIn: true,
      },
      android: {
        adaptiveIcon: {
          foregroundImage: "./assets/adaptive-icon.png",
          backgroundColor: "#ffffff",
        },
        package: "com.kernalan.lifesage",
        versionCode: 1,
      },
      web: {
        favicon: "./assets/favicon.png",
        bundler: "metro",
      },
      plugins: [
        "expo-router",
        [
          "expo-build-properties",
          {
            ios: {
              useFrameworks: "static",
            },
          },
        ],
      ],
      extra: {
        eas: {
          projectId: "ad20d820-991c-44ec-8528-26af05ab9834",
        },
        FIREBASE_API_KEY: process.env.FIREBASE_API_KEY,
        FIREBASE_AUTH_DOMAIN: process.env.FIREBASE_AUTH_DOMAIN,
        FIREBASE_PROJECT_ID: process.env.FIREBASE_PROJECT_ID,
        FIREBASE_STORAGE_BUCKET: process.env.FIREBASE_STORAGE_BUCKET,
        FIREBASE_MESSAGING_SENDER_ID: process.env.FIREBASE_MESSAGING_SENDER_ID,
        FIREBASE_APP_ID: process.env.FIREBASE_APP_ID,
        FIREBASE_MEASUREMENT_ID: process.env.FIREBASE_MEASUREMENT_ID,
        API_BASE_URL: process.env.API_BASE_URL,
      },
      owner: "kernalan",
    },
  };
};
```

I haven't been able to test more, but I suspect EAS expects an object, not a function wrapper. I used the function to be able to do dynamic swapping of env variables here, but getting rid of that approach fixes the issue.

--- a/packages/expo-updates/CHANGELOG.md
+++ b/packages/expo-updates/CHANGELOG.md
@@ -19,6 +19,7 @@
 - [iOS] Rollback to system SQLite3 and fix incompatible issue when any third-party library uses iOS system SQLite3. ([#30826](https://github.com/expo/expo/pull/30826) by [@kudo](https://github.com/kudo))
 - Use expo-updates as source of truth for runtime version in dev client ([#31453](https://github.com/expo/expo/pull/31453) by [@wschurman](https://github.com/wschurman))
 - Fixed iOS reload crash on New Architecture mode. ([#31789](https://github.com/expo/expo/pull/31789) by [@kudo](https://github.com/kudo))
+- [cli] Disable console during runtime and fingeprint evaluation ([#31874](https://github.com/expo/expo/pull/31874) by [@wschurman](https://github.com/wschurman))
 
 ### 💡 Others
 

--- a/packages/expo-updates/cli/build/configureCodeSigning.js
+++ b/packages/expo-updates/cli/build/configureCodeSigning.js
@@ -59,7 +59,7 @@ Configure expo-updates code signing for this project and verify setup
     const certificateInput = (0, args_1.requireArg)(args, '--certificate-input-directory');
     const keyInput = (0, args_1.requireArg)(args, '--key-input-directory');
     const keyid = args['--keyid'];
-    return await configureCodeSigningAsync((0, args_1.getProjectRoot)(args), {
+    await configureCodeSigningAsync((0, args_1.getProjectRoot)(args), {
         certificateInput,
         keyInput,
         keyid,

--- a/packages/expo-updates/cli/build/generateCodeSigning.js
+++ b/packages/expo-updates/cli/build/generateCodeSigning.js
@@ -63,7 +63,7 @@ Generate expo-updates private key, public key, and code signing certificate usin
     const certificateOutput = (0, args_1.requireArg)(args, '--certificate-output-directory');
     const certificateValidityDurationYears = (0, args_1.requireArg)(args, '--certificate-validity-duration-years');
     const certificateCommonName = (0, args_1.requireArg)(args, '--certificate-common-name');
-    return await generateCodeSigningAsync((0, args_1.getProjectRoot)(args), {
+    await generateCodeSigningAsync((0, args_1.getProjectRoot)(args), {
         certificateValidityDurationYears,
         keyOutput,
         certificateOutput,

--- a/packages/expo-updates/cli/build/generateFingerprint.js
+++ b/packages/expo-updates/cli/build/generateFingerprint.js
@@ -32,6 +32,7 @@ const chalk_1 = __importDefault(require("chalk"));
 const args_1 = require("./utils/args");
 const errors_1 = require("./utils/errors");
 const Log = __importStar(require("./utils/log"));
+const withConsoleDisabledAsync_1 = require("./utils/withConsoleDisabledAsync");
 const generateFingerprint = async (argv) => {
     const args = (0, args_1.assertArgs)({
         // Types
@@ -71,14 +72,18 @@ Generate fingerprint for use in expo-updates runtime version
     }
     const debug = args['--debug'];
     const projectRoot = (0, args_1.getProjectRoot)(args);
-    let result;
-    try {
-        const workflow = workflowArg ?? (await resolveWorkflowAsync(projectRoot, platform));
-        result = await createFingerprintAsync(projectRoot, platform, workflow, { silent: true, debug });
-    }
-    catch (e) {
-        throw new errors_1.CommandError(e.message);
-    }
+    const result = await (0, withConsoleDisabledAsync_1.withConsoleDisabledAsync)(async () => {
+        try {
+            const workflow = workflowArg ?? (await resolveWorkflowAsync(projectRoot, platform));
+            return await createFingerprintAsync(projectRoot, platform, workflow, {
+                silent: true,
+                debug,
+            });
+        }
+        catch (e) {
+            throw new errors_1.CommandError(e.message);
+        }
+    });
     console.log(JSON.stringify(result));
 };
 exports.generateFingerprint = generateFingerprint;

--- a/packages/expo-updates/cli/build/resolveRuntimeVersion.js
+++ b/packages/expo-updates/cli/build/resolveRuntimeVersion.js
@@ -32,6 +32,7 @@ const chalk_1 = __importDefault(require("chalk"));
 const args_1 = require("./utils/args");
 const errors_1 = require("./utils/errors");
 const Log = __importStar(require("./utils/log"));
+const withConsoleDisabledAsync_1 = require("./utils/withConsoleDisabledAsync");
 const resolveRuntimeVersion = async (argv) => {
     const args = (0, args_1.assertArgs)({
         // Types
@@ -67,18 +68,19 @@ Resolve expo-updates runtime version
         throw new errors_1.CommandError(`Invalid workflow argument: ${workflow}. Must be either 'managed' or 'generic'`);
     }
     const debug = args['--debug'];
-    let runtimeVersionInfo;
-    try {
-        runtimeVersionInfo = await resolveRuntimeVersionAsync((0, args_1.getProjectRoot)(args), platform, {
-            silent: true,
-            debug,
-        }, {
-            workflowOverride: workflow,
-        });
-    }
-    catch (e) {
-        throw new errors_1.CommandError(e.message);
-    }
+    const runtimeVersionInfo = await (0, withConsoleDisabledAsync_1.withConsoleDisabledAsync)(async () => {
+        try {
+            return await resolveRuntimeVersionAsync((0, args_1.getProjectRoot)(args), platform, {
+                silent: true,
+                debug,
+            }, {
+                workflowOverride: workflow,
+            });
+        }
+        catch (e) {
+            throw new errors_1.CommandError(e.message);
+        }
+    });
     console.log(JSON.stringify(runtimeVersionInfo));
 };
 exports.resolveRuntimeVersion = resolveRuntimeVersion;

--- a/packages/expo-updates/cli/build/utils/withConsoleDisabledAsync.d.ts
+++ b/packages/expo-updates/cli/build/utils/withConsoleDisabledAsync.d.ts
@@ -1,0 +1,1 @@
+export declare function withConsoleDisabledAsync<T>(block: () => Promise<T>): Promise<T>;

--- a/packages/expo-updates/cli/build/utils/withConsoleDisabledAsync.js
+++ b/packages/expo-updates/cli/build/utils/withConsoleDisabledAsync.js
@@ -1,0 +1,24 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.withConsoleDisabledAsync = void 0;
+async function withConsoleDisabledAsync(block) {
+    const loggingFunctions = {
+        log: console.log,
+        warn: console.warn,
+        error: console.error,
+    };
+    // Disable logging for this command since the only thing printed to stdout should be the JSON output.
+    console.log = function () { };
+    console.warn = function () { };
+    console.error = function () { };
+    try {
+        return await block();
+    }
+    finally {
+        // Re-enable logging functions for testing.
+        console.log = loggingFunctions.log;
+        console.warn = loggingFunctions.warn;
+        console.error = loggingFunctions.error;
+    }
+}
+exports.withConsoleDisabledAsync = withConsoleDisabledAsync;

--- a/packages/expo-updates/cli/src/configureCodeSigning.ts
+++ b/packages/expo-updates/cli/src/configureCodeSigning.ts
@@ -43,7 +43,7 @@ Configure expo-updates code signing for this project and verify setup
   const keyInput = requireArg(args, '--key-input-directory');
   const keyid = args['--keyid'];
 
-  return await configureCodeSigningAsync(getProjectRoot(args), {
+  await configureCodeSigningAsync(getProjectRoot(args), {
     certificateInput,
     keyInput,
     keyid,

--- a/packages/expo-updates/cli/src/generateCodeSigning.ts
+++ b/packages/expo-updates/cli/src/generateCodeSigning.ts
@@ -50,7 +50,7 @@ Generate expo-updates private key, public key, and code signing certificate usin
   );
   const certificateCommonName = requireArg(args, '--certificate-common-name');
 
-  return await generateCodeSigningAsync(getProjectRoot(args), {
+  await generateCodeSigningAsync(getProjectRoot(args), {
     certificateValidityDurationYears,
     keyOutput,
     certificateOutput,

--- a/packages/expo-updates/cli/src/generateFingerprint.ts
+++ b/packages/expo-updates/cli/src/generateFingerprint.ts
@@ -5,6 +5,7 @@ import { Command } from './cli';
 import { requireArg, assertArgs, getProjectRoot } from './utils/args';
 import { CommandError } from './utils/errors';
 import * as Log from './utils/log';
+import { withConsoleDisabledAsync } from './utils/withConsoleDisabledAsync';
 
 export const generateFingerprint: Command = async (argv) => {
   const args = assertArgs(
@@ -60,13 +61,17 @@ Generate fingerprint for use in expo-updates runtime version
 
   const projectRoot = getProjectRoot(args);
 
-  let result;
-  try {
-    const workflow = workflowArg ?? (await resolveWorkflowAsync(projectRoot, platform));
-    result = await createFingerprintAsync(projectRoot, platform, workflow, { silent: true, debug });
-  } catch (e: any) {
-    throw new CommandError(e.message);
-  }
+  const result = await withConsoleDisabledAsync(async () => {
+    try {
+      const workflow = workflowArg ?? (await resolveWorkflowAsync(projectRoot, platform));
+      return await createFingerprintAsync(projectRoot, platform, workflow, {
+        silent: true,
+        debug,
+      });
+    } catch (e: any) {
+      throw new CommandError(e.message);
+    }
+  });
 
   console.log(JSON.stringify(result));
 };

--- a/packages/expo-updates/cli/src/resolveRuntimeVersion.ts
+++ b/packages/expo-updates/cli/src/resolveRuntimeVersion.ts
@@ -5,6 +5,7 @@ import { Command } from './cli';
 import { requireArg, assertArgs, getProjectRoot } from './utils/args';
 import { CommandError } from './utils/errors';
 import * as Log from './utils/log';
+import { withConsoleDisabledAsync } from './utils/withConsoleDisabledAsync';
 
 export const resolveRuntimeVersion: Command = async (argv) => {
   const args = assertArgs(
@@ -57,21 +58,23 @@ Resolve expo-updates runtime version
 
   const debug = args['--debug'];
 
-  let runtimeVersionInfo;
-  try {
-    runtimeVersionInfo = await resolveRuntimeVersionAsync(
-      getProjectRoot(args),
-      platform,
-      {
-        silent: true,
-        debug,
-      },
-      {
-        workflowOverride: workflow,
-      }
-    );
-  } catch (e: any) {
-    throw new CommandError(e.message);
-  }
+  const runtimeVersionInfo = await withConsoleDisabledAsync(async () => {
+    try {
+      return await resolveRuntimeVersionAsync(
+        getProjectRoot(args),
+        platform,
+        {
+          silent: true,
+          debug,
+        },
+        {
+          workflowOverride: workflow,
+        }
+      );
+    } catch (e: any) {
+      throw new CommandError(e.message);
+    }
+  });
+
   console.log(JSON.stringify(runtimeVersionInfo));
 };

--- a/packages/expo-updates/cli/src/utils/withConsoleDisabledAsync.ts
+++ b/packages/expo-updates/cli/src/utils/withConsoleDisabledAsync.ts
@@ -1,0 +1,20 @@
+export async function withConsoleDisabledAsync<T>(block: () => Promise<T>): Promise<T> {
+  const loggingFunctions = {
+    log: console.log,
+    warn: console.warn,
+    error: console.error,
+  };
+  // Disable logging for this command since the only thing printed to stdout should be the JSON output.
+  console.log = function () {};
+  console.warn = function () {};
+  console.error = function () {};
+
+  try {
+    return await block();
+  } finally {
+    // Re-enable logging functions for testing.
+    console.log = loggingFunctions.log;
+    console.warn = loggingFunctions.warn;
+    console.error = loggingFunctions.error;
+  }
+}

